include/seqan3/alphabet/detail/alphabet_proxy.hpp

```diff
@@ -247,17 +184,40 @@ class alphabet_proxy : public alphabet_base<derived_type,
     //!\brief Implicit conversion to the emulated type.
     constexpr operator alphabet_type() const noexcept
     {
-        return assign_rank_to(to_rank(), alphabet_type{});
+        if constexpr (std::is_class_v<alphabet_type>)
+            return *this;
+        else
+            return assign_rank_to(base_t::to_rank(), alphabet_type{});
+
+        /* This is used explicitly in this class often to prevent considering alphabet_type's constructors
```

Comment:
This sentence makes no sense to me oO
Also, why don't you put this into the details of the documentation of the implicit conversion?